org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporter.java

```diff
@@ -174,7 +176,12 @@ private void updateProjects(Collection<IProject> projects, long lastWorkspaceSta
 		while (iterator.hasNext()) {
 			IProject project = iterator.next();
 			project.open(monitor);
-			project.refreshLocal(IResource.DEPTH_INFINITE, monitor);
+			if (Platform.OS_WIN32.equals(Platform.getOS())) {
+				project.refreshLocal(IResource.DEPTH_ONE, monitor);
+				((Workspace) ResourcesPlugin.getWorkspace()).getRefreshManager().refresh(project);
```

Comment:
@fbricon This line will refresh the project and detect the changes after initialization. And it is similar to eclipse's behavior  for refreshing.